cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -454,6 +471,40 @@ func generateMountsCommonInstallRecover(mst *initramfsMountsState) error {
 	return sysconfig.ConfigureTargetSystem(configOpts)
 }
 
+func isSavePresent() (bool, error) {
+	devPath := filepath.Join(dirs.GlobalRootDir, "/dev/disk/by-partlabel/ubuntu-save")
+	_, err := os.Stat(devPath)
+	if err != nil && !os.IsNotExist(err) {
+		return false, err
+	}
+	return err == nil, nil
+}
+
+func maybeMountSave(disk disks.Disk, encrypted bool, mountOpts *systemdMountOptions) (haveSave bool, err error) {
+	haveSave, err = isSavePresent()
+	if err != nil {
+		return false, err
+	}
+	if !haveSave {
+		return false, nil
+	}
+
+	// we have ubuntu-save, try to mount it
+	const lockKeysOnFinish = true
```

Comment:
we can't have this set to true for both data and save, only for the last one of the two